BDArmory/Damage/HitpointTracker.cs

```diff
@@ -563,13 +568,13 @@ IEnumerator DelayedOnStart()
                 }
             }
             if (!isProcWing) //moving this here so any dynamic texture adjustment post spawn (TURD/TUFX/etc) will be grabbed by the defaultShader census
-            {
+            {                   //have this be done by RadarUtils when doing RCS snapshots instead?
                 var r = part.GetComponentsInChildren<Renderer>();
                 for (int i = 0; i < r.Length; i++)
                 {
                     if (r[i].GetComponentInParent<Part>() != part) continue; // Don't recurse to child parts.
                     int key = r[i].material.GetInstanceID(); // The instance ID is unique for each object (not just component or gameObject).
-                    defaultShader.Add(key, r[i].material.shader);
+                    defaultShader.Add(key, r[i].material.shader); //This doesn't grab part variants - parts with variants that are switched to will not register in defaultShader, and render as black in the RCS window
```

Comment:
I'm seeing some of these exceptions
```
[EXC 22:27:28.232] ArgumentException: An item with the same key has already been added. Key: -1209990
  System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) (at <9577ac7a62ef43179789031239ba8798>:0)
  System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) (at <9577ac7a62ef43179789031239ba8798>:0)
  BDArmory.Damage.HitpointTracker+<DelayedOnStart>d__100.MoveNext () (at <8074436c38894f3ab224691a173038e3>:0)
```
in the current RWP DLL. I suspect it's coming from here since there's no check for `key` already being in the dictionary.